src/sentry/snuba/metrics/fields/snql.py

```diff
@@ -304,3 +305,40 @@ def session_duration_filters(org_id):
             ),
         )
     ]
+
+
+def histogram(
+    aggregate_filter,
+    histogram_buckets: int = 100,
+    histogram_from: Optional[float] = None,
+    histogram_to: Optional[float] = None,
+    alias=None,
+):
+    zoom_conditions = zoom_histogram(
+        histogram_buckets=histogram_buckets,
+        histogram_from=histogram_from,
+        histogram_to=histogram_to,
+    )
+    if zoom_conditions is not None:
+        conditions = Function("and", [zoom_conditions, aggregate_filter])
+    else:
+        conditions = aggregate_filter
+
+    return Function(
+        "histogramIf(250)",
+        [Column("value"), conditions],
+        alias=alias,
+    )
+
+
+def rate(aggregate_filter, numerator, denominator=None, alias=None):
+    if denominator is None:
+        denominator = 1.0
```

Comment:
why not just to put 1.0 as default kwarg value?
This pattern `if denominator is None:` typically we use when argument is muttable e.g. list.

Reply:
agreed